pyscriptjs/src/utils.ts

```diff
@@ -111,3 +101,54 @@ export function getAttribute(el: Element, attr: string): string | null {
     }
     return null;
 }
+
+export function _createAlertBanner(message: string, level: "error" | "warning" = "error", logMessage = true) {
+
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+    switch(`log-${level}-${logMessage}`) {
+        case "log-error-true":
+        console.error(message);
+        break;
+        case "log-warning-true":
+        console.warn(message)
+        break;
+    }
+
+    const banner = document.createElement("div")
+    banner.className = `alert-banner py-${level}`
+    banner.innerHTML = message
+
+    if (level === "warning") {
+        const closeButton = document.createElement("button")
+
+        closeButton.id = "alert-close-button"
+        closeButton.addEventListener("click", () => {
+            banner.remove()
+        })
+        closeButton.innerHTML = CLOSEBUTTON
+
+        banner.appendChild(closeButton)
+    }
+
+    document.body.prepend(banner)
+}
+
+export function withUserErrorHandler(fn) {
```

Comment:
this function deserves a docstring IMHO :).

But also, I wonder whether it should go to e.g. `exceptions.ts`, together with `showWarning` and `_createAlertBanner`.
Basically, I think that we should stop using `utils.ts` as a confusing aggregate of unrelated functions and use it only for the few functions which are real "utilities".

Reply:
Originally I was thinking to put it in exceptions, but then wondered if it was the right place for it haha
I'll move them and add the docstring 😄